File: backend/src/model/VirtualMachine.go
package model

type VirtualMachine struct {
    Id     int    `json:"id,omitempty"`
	Name  string  `json:"name,omitempty"`
	Cpu   int     `json:"cpu,omitempty"`
	Ram   int     `json:"ram,omitempty"`
	Disk  int     `json:"disk,omitempty"`
	DiskType   int     `json:"disk_type,omitempty"`
	Volumes   VirtualVolumes `json:"volumes,omitempty"`
}

type VirtualMachineDB struct {
    Id     int    `json:"id,omitempty"`
	Name  string  `json:"name,omitempty"`
	Cpu   int     `json:"cpu,omitempty"`
	Ram   int     `json:"ram,omitempty"`
	Disk  int     `json:"disk,omitempty"`
	DiskType   int     `json:"disk_type,omitempty"`
	Volumes   string     `json:"volumes,omitempty"`
}

type VirtualMachines []VirtualMachine

File: frontend/src/app/modules/virtual-machine/store/models/virtual-machine.ts
import { environment } from "src/environments/environment";

export interface VirtualMachine {
    id: number | null;
    name: string | null;
	cpu: number | null;
	ram: number | null;
	disk: number | null;
	disk_type: number | null;
}

export class VirtualMachineModel implements VirtualMachine
{
	id: number | null = null;
    name: string | null = null;
	cpu: number | null = null;
	ram: number | null = null;
	disk: number | null = null;
	disk_type: number | null = null;

	diskTypes: { [key: number]: string; } = environment.config.virtialMachine.disk_type;

	get diskType(): string {
		return this.disk_type ? this.diskTypes[this.disk_type] : ' - ' + this.disk_type + '-';
	}
}